library/std/src/sys/pal/solid/fs.rs

```diff
@@ -325,14 +377,17 @@ fn cstr(path: &Path) -> io::Result<CString> {
 
 impl File {
     pub fn open(path: &Path, opts: &OpenOptions) -> io::Result<File> {
```

Comment:
This still still inconsistent: why is `open` kept in some targets but removed in other? I would expect everything to use `open_native` only and for `open` to be removed (or possible just change the signature of `open` to take native path).